src/Plugin/media/Source/Embeddable.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Drupal\stanford_media\Plugin\media\Source;
+
+use Drupal\media\Plugin\media\Source\OEmbed;
+use Drupal\media\MediaInterface;
+use Drupal\media\MediaTypeInterface;
+
+/**
+ * @MediaSource(
+ *   id = "embeddable",
+ *   label = @Translation("Stanford Embedded Media"),
+ *   description = @Translation("Embeds a third-party resource."),
+ *   default_thumbnail_filename = "generic.png",
+ *   providers = {"ArcGIS StoryMaps", "CircuitLab", "Codepen", "Dailymotion", "Facebook", "Flickr", "Getty Images", "Instagram", "Issuu", "Livestream", "MathEmbed", "Simplecast", "SlideShare", "SoundCloud", "Spotify", "Stanford Digital Repository", "Twitter"},
+ *   allowed_field_types = {"string", "string_long"},
+ * )
+ */
+class Embeddable extends OEmbed {
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $attribute_name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getMetadata(MediaInterface $media, $name) {
+    if ($this->hasUnstructured($media)) {
+      return $this->getUnstructuredMetadata($media, $name);
+    }
+    return parent::getMetadata($media, $name);
+  }
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   * This is an alternate version to account for unstructured embeds.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $attribute_name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getUnstructuredMetadata(MediaInterface $media, $name) {
+    switch ($name) {
+      case 'default_name':
+        if ($title = $this->getMetadata($media, 'title')) {
+          return $title;
+        }
+        elseif ($url = $this->getMetadata($media, 'url')) {
+          return $url;
+        }
+        return parent::getMetadata($media, 'default_name');
+
+      case 'thumbnail_uri':
+        parent::getMetadata($media, 'thumbnail_uri');
```

Comment:
```suggestion
        parent::getMetadata($media, 'thumbnail_uri');
        break;
```

Reply:
actually, I think what I need there is `return parent::getMetadata($media, 'thumbnail_uri');`.  Fixed.